src/main/java/com/google/starfish/ConnectionPoolContextListener.java

```diff
@@ -0,0 +1,195 @@
+    // live longer than this many milliseconds will be closed and reestablished between uses.
+    config.setMaxLifetime(MINUTE * 30);
+
+    // Initialize the connection pool using the configuration object.
+    DataSource pool = new HikariDataSource(config);
+
+    return pool;
+  }
+
+  private void createLabelsTable(DataSource pool) throws SQLException {
+    try (Connection conn = pool.getConnection()) {
+      String stmt =
+          "CREATE TABLE IF NOT EXISTS labels ( "
+              + "title VARCHAR(255) PRIMARY KEY,"
+              + "type ENUM('School', 'Course', 'Other') NOT NULL );";
+      try (PreparedStatement createTableStatement = conn.prepareStatement(stmt)) {
+        createTableStatement.execute();
+      }
+    }
+  }
+
+  private void createUsersTable(DataSource pool) throws SQLException {
+    try (Connection conn = pool.getConnection()) {
+      String stmt =
+          "CREATE TABLE IF NOT EXISTS users ( "
+              + "id INT PRIMARY KEY AUTO_INCREMENT,"
+              + "display_picture VARCHAR(255),"
+              + "display_name VARCHAR(255),"
+              + "date_joined DATE,"
+              + "email VARCHAR(255) NOT NULL,"
+              + "points INT NOT NULL,"
+              + "school VARCHAR(255),"
+              + "FOREIGN KEY (`school`)"
+                  + "REFERENCES `labels` (`title`)"
+                  + "ON UPDATE RESTRICT ON DELETE CASCADE );";
+      try (PreparedStatement createTableStatement = conn.prepareStatement(stmt)) {
+        createTableStatement.execute();
+      }
+    }
+  }
+  
+  private void createNotesTable(DataSource pool) throws SQLException {
+    try (Connection conn = pool.getConnection()) {
+      String stmt =
+          "CREATE TABLE IF NOT EXISTS notes ( "
+              + "id INT PRIMARY KEY AUTO_INCREMENT,"
+              + "author_id INT NOT NULL,"
+              + "school VARCHAR(255) NOT NULL,"
+              + "course VARCHAR(255) NOT NULL,"
+              + "title VARCHAR(255) NOT NULL,"
+              + "source_url VARCHAR(255) NOT NULL,"
+              + "pdf_source VARCHAR(255),"
+              + "date_created DATE NOT NULL,"
+              + "num_downloads INT NOT NULL,"
+              + "FOREIGN KEY (`author_id`)"
+                  + "REFERENCES `users` (`id`)"
+                  + "ON DELETE CASCADE,"
+              + "FOREIGN KEY (`school`)"
+                  + "REFERENCES `labels` (`title`),"
+              + "FOREIGN KEY (`course`)"
+                  + "REFERENCES `labels` (`title`),"
+              + "CONSTRAINT pdf_or_source_url_not_null check ("
+                  + "`source_url` IS NOT NULL OR `pdf_source` IS NOT NULL) );";
+      try (PreparedStatement createTableStatement = conn.prepareStatement(stmt)) {
+        createTableStatement.execute();
+      }
+    }
+  }
+
+  private void createOtherNoteLabelsTable(DataSource pool) throws SQLException {
```

Comment:
Does 'other' mean non-default here?

Other feels a bit too ambiguous, would prefer a rename to something like non-default.

Reply:
I don't think other in this sense exactly means 'non-default'. What it is is that all notes have 2 required labels, which are the school and the course, and then this table represents the bridge between the many to many relationship between the notes and labels for what I called 'other' labels, i.e. labels like 'Physics', 'Chemistry', 'AI', 'Calculus', etc.
Do you think 'optional' would be better, so it would be the 'optional_note_labels' table?

Reply:
Or Miscellaneous, maybe.

Reply:
miscLabels definitely sounds better than other and still more descriptive than optional. So I'm thinking I'll push 'createMiscNoteLabelsTable' then?

Reply:
Sounds good to me, I can start calling them that in the frontend, too.

Reply:
Sounds good!